lumicks/pylake/channel.py

```diff
@@ -158,7 +158,7 @@ def to_index(t):
             """Convert a timestamp into a continuous channel index (assumes t >= self.start)"""
             return (t - self.start + self.dt - 1) // self.dt
 
-        start = max(start, self.start)
+        start = max(start + np.mod(start, self.dt), self.start)
```

Comment:
You can use the builtin `%` operator here since it's faster for scalars.

Reply:
Ah yes, makes sense. Thanks!